@here/harp-mapview/lib/Utils.ts

```diff
@@ -944,78 +944,77 @@ export namespace MapViewUtils {
-
-        const cameraToTarget = cache.vector3[1];
-        cameraToTarget.copy(cameraPos).sub(worldTarget).negate();
-
-        const cameraToTargetNormalized = new THREE.Vector3().copy(cameraToTarget).normalize();
-
-        const offsetVector = new THREE.Vector3();
-
-        const cameraToPointOnRefPlane = new THREE.Vector3();
-        const pointOnRefPlane = new THREE.Vector3();
-
-        function checkAngle(
-            point: THREE.Vector3,
-            referencePlane: THREE.Plane,
-            maxAngle: number,
-            fovFactor: number
-        ) {
-            referencePlane.projectPoint(point, pointOnRefPlane);
-            cameraToPointOnRefPlane.copy(cameraPos).sub(pointOnRefPlane).negate();
-
-            const viewAngle = cameraToTarget.angleTo(cameraToPointOnRefPlane);
-
-            if (viewAngle <= maxAngle) {
-                return;
-            }
-
-            const cameraToPointLen = cameraToPointOnRefPlane.length();
-            const cameraToTargetLen = cameraToTarget.length();
-
-            const newCameraDistance =
-                cameraToPointLen * (Math.sin(viewAngle) * fovFactor - Math.cos(viewAngle)) +
-                cameraToTargetLen;
-
-            offsetVector
-                .copy(cameraToTargetNormalized)
-                .multiplyScalar(cameraToTargetLen - newCameraDistance);
-
-            cameraPos.add(offsetVector);
-            cameraToTarget.sub(offsetVector);
-        }
-
+        //                camY
+        //       targetDist^
+        //      |<-------->|     Ps
+        //     constD pEyeZ|    /|  ^
+        //      |<-->|<--->|   / |  |
+        //      |    |     |  /  |  | |ndcY-O.y|*h/2
+        //      |    |     | /   |  |
+        //  <---T----P'----C0----O  v
+        // camZ      |_|  /|     |
+        //           |   / |<--->|
+        //      PcamY|  /     f
+        //           | / (focal length)
+        //           |/
+        //           P
+        //                            camY
+        //     constD      newPEyeZ    ^          Ps
+        //      |<-->|<--------------->|       _-`|  ^
+        //      |    |                 |    _-`   |  | |sign(ndcY)-O.y|h/2
+        //      |    |                 | _-`      |  |
+        //  <---T----P'----C0----------C1---------O  v
+        // camZ      |_|            _-`|          |              C0  - Initial camera position
+        //           |           _-`   |<-------->|              C1  - New camera position
+        //      PcamY|        _-`           f                    T   - Camera target
+        //           |     _-`        (focal length)             P   - Bounds point (world space)
+        //           |  _-`                                      Ps  - P projected on screen.
+        //           P-`                                         O   - Principal point.
+        //
+        // Diagram showing how to calculate the camera distance on the camera YZ plane so that a
```

Comment:
I didn't understand the diagrams till I had read the description, and even then it wasn't fully clear.

They should have some title, and the second diagram should have some description that highlights that we are computing the factor to move C from C0 to C1. Also, it should be highlighted that Ps is on the edge of the screen. I know it is described later, but it would be good to have it documented here too.

Reply:
done